apps/devhub/tasks.py

```diff
@@ -54,6 +55,25 @@ def validate(file_, listed=None):
         return result
 
 
+def validate_and_submit(addon, file_, listed=None):
+    return validate(file_, listed=listed,
+                    callback=submit_file.si(addon.pk, file_.pk))
+
+
+@task
+def submit_file(addon_pk, file_pk):
+    addon = Addon.unfiltered.get(pk=addon_pk)
+    file_ = FileUpload.objects.get(pk=file_pk)
+    validation = json.loads(file_.validation)
+    if (file_.automated_signing and validation['passed_auto_validation']) or (
+            not file_.automated_signing and file_.valid):
+        # Import loop.
+        from devhub.views import auto_sign_version
+
```

Comment:
`Version.from_upload` does have some logging but it would probably be good to see it coming from the new API. Good call. Should I be adding any metrics too?